server.js

```diff
@@ -47,74 +49,188 @@ app.get("/", (req, res) => {
-      res.json(journals);
+      // user does not exist or either username, email or password are not valid.
+      if (!journals.length) {
+        res.sendStatus(404);
+      } else {
+        res.json(journals);
+      }
     })
     .catch((err) => {
-      res.send("Error " + err);
+      console.log(err);
+      res.sendStatus(500);
     });
 });
-
+// Not sure if this will be needed
 app.post("/logout", (req, res) => {
   res.send("logout");
 });
 
-app.post("/register", (req, res) => {
+app.post("/register", async (req, res) => {
   const { user_name, email, pass_word } = req.body;
-  //TODO: user_name and email need to be unique
-  /*
-   If the user or email exist send a 409, if another error occurs, send a 500.
-   Send a 200 if everything is ok.
-  */
-  Users.create({ user_name, email, pass_word }).then(()=> {
-    res.sendStatus(200);
-  }).catch(err => {
-    res.send(500);
-  })
+
+  // Checks user_name does not exist
+  const isUsernameInUse = await Users.findAll({ where: { user_name } });
   
+  if (isUsernameInUse.length) {
+    return res.sendStatus(409);
+  }
+  // Check email is not being used
+  const isEmailInUse = await Users.findAll({ where: { email } });
+ 
+  if (isEmailInUse.length) {
+    return res.sendStatus(409);
+  }
+  // Create the user
+  Users.create({ user_name, email, pass_word })
+    .then(() => {
+      res.sendStatus(200);
+    })
+    .catch((err) => {
+      res.send(500);
+    });
+});
+
+app.get("/journals/:userId/", (req, res) => {
+  Journals.findAll({ where: { user_id: req.params.userId } })
+    .then((journals) => {
+      res.json(journals);
+    })
+    .catch((err) => {
+      res.sendStatus(500);
+    });
 });
 
 app
-  .route("/journals/:userId")
-  .get((req, res) => {
-    Journals.findAll()
-      .then((journals) => {
-        res.json(journals);
+  .route("/journals/:userId/:strategyId")
+  .post((req, res) => {
+    const payload = {
+      user_id: req.params.userId,
+      strategy_id: req.params.strategyId,
+      pair: req.body.pair,
+      comments: req.body.comments,
+      order_type: req.body.order_type,
+      pips_gained_lost: req.body.pips_gained_lost,
+      img_link: req.body.img_link,
+    };
+
+    Journals.create(payload)
+      .then(() => {
+        res.sendStatus(200);
       })
       .catch((err) => {
-        res.send("Error " + err);
+        res.send(err);
       });
   })
-  .post((req, res) => {
-    res.send("create a journal");
-  })
   .put((req, res) => {
-    res.send("update a journal");
+    const strategy_id = req.params.strategyId;
+    const user_id = req.params.userId;
+    const { journal_id } = req.body;
+    const updatedPayload = {
+      pair: req.body.pair,
+      comments: req.body.comments,
+      order_type: req.body.order_type,
+      pips_gained_lost: req.body.pips_gained_lost,
+      img_link: req.body.img_link,
+    };
+
+    Journals.update(updatedPayload, {
+      where: { journal_id, strategy_id, user_id },
```

Comment:
This only work in the case that those properties have a value. If they're `undefined`, this will produce an error.